client/src/amplify-ui/cli-feature-flag/feature-flags.json

```
@@ -143,21 +143,105 @@
+        "type": "Feature",
+        "valueType": "Boolean",
+        "versionAdded": "4.40.0",
+        "deprecationDate": "May 1st 2021",
+        "values": [
+          {
+            "value": "true",
+            "description": "Allows using newly migrated docsgen package.",
+            "defaultNewProject": true,
+            "defaultExistingProject": false
+          },
+          {
+            "value": "false",
+            "description": "Allows using older docsgen package",
+            "defaultNewProject": false,
+            "defaultExistingProject": true
+          }
+        ]
+      },
+      "useTypesGeneratorPlugin": {
+        "description": "Allows to use the migrated graphQL types generator package at https://github.com/aws-amplify/amplify-codegen",
+        "type": "Feature",
+        "valueType": "Boolean",
+        "versionAdded": "4.40.0",
+        "deprecationDate": "May 1st 2021",
+        "values": [
+          {
+            "value": "true",
+            "description": "Allows using newly migrated typesgen package.",
+            "defaultNewProject": true,
+            "defaultExistingProject": false
+          },
+          {
+            "value": "false",
+            "description": "Allows using older typesgen package",
+            "defaultNewProject": false,
+            "defaultExistingProject": true
+          }
+        ]
+      },
+      "cleanGeneratedModelsDirectory": {
+        "description": "Clears the generated models folder before they are re-generated.",
+        "type": "Feature",
+        "valueType": "Boolean",
+        "versionAdded": "4.40.0",
+        "deprecationDate": "May 1st 2021",
+        "values": [
+          {
+            "value": "true",
+            "description": "Recommended to be set true when there are no hand-written files in the modelgen output folder.",
+            "defaultNewProject": true,
+            "defaultExistingProject": false
+          },
+          {
+            "value": "false",
+            "description": "Does not remove pre-existing files from the modelgen output folder.",
+            "defaultNewProject": false,
+            "defaultExistingProject": true
+          }
+        ]
+      },
+      "retainCaseStyle": {
+        "description": "Retains the case style developer uses while naming the graphQL types in schema",
+        "type": "Feature",
+        "valueType": "Boolean",
+        "versionAdded": "4.40.0",
+        "deprecationDate": "May 1st 2021",
+        "values": [
+          {
+            "value": "true",
+            "description": "Set true to retain the different case styles used for type names in the schema. Refer https://github.com/aws-amplify/amplify-codegen/pull/89",
+            "defaultNewProject": true,
+            "defaultExistingProject": false
+          },
+          {
+            "value": "false",
+            "description": "Keeps the already generated variable names to avoid breaking changes",
```

```suggestion
            "description": "Keeps the generated variable names to avoid breaking changes",
```

How does this avoid breaking changes?